src/macros.rs

```diff
@@ -13,11 +13,97 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-macro_rules! impl_marker {
-    ($M:ty; $($T:ty),* $(,)*) => {
-        $(impl $M for $T { })+
-    }
-}
+#[macro_export]
+macro_rules! impl_marker(
```

Comment:
Would it be possible to add comments telling what each part of the macro does ? It is very hard to understand anything just by reading the code.